### PR TITLE
Create CI/CD pipeline with GitHub Actions for developer-assessment coding test

### PR DESCRIPTION
**Pull request by pradhanangan**
JIRA: https://clearpoint.atlassian.net/browse/ENP-293

__Changes__
Created simple CI/CD pipeline using github actions to build and run unit test Backend and Frontend application. Currently it run on every push request.
Dockerized the Backend application

__New File:__
.github/workflows/main.yaml
Frontend/Dockerfile
tasks (Bash file)

__Modified File:__
Frontend/package.json
The line was added because testing Frontend app using jest was throwing error
`Jest failed to parse a file. This happens e.g. when your code or its dependencies use non-standard JavaScript syntax, or when Jest is not configured to support such syntax.`
and axios node module uses non standard Javascript syntax.

### CODE REVIEW

File: .github/workflows/main.yaml [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+      - name: Extract branch name
+        shell: bash
+        run: echo "##[set-output name=branch;]$(echo ${GITHUB_REF#refs/heads/})"
+        id: extract_branch
+      - uses: actions/upload-artifact@v3
+        with:
+          name: imagerc-${{ steps.extract_branch.outputs.branch }}-${{ github.sha }}-${{ github.run_number }}-${{ github.run_attempt }}-${{ github.workflow }}
+          path: Backend/TodoList.Api/imagerc.yml
+          retention-days: 5
+      - run: ./tasks e2e
```
Comment (mhurbuns) [inline .github/workflows/main.yaml:40]: e2e is not needed

File: tasks [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+DOCKER_OPS_VERSION="2.0.0"
+DOCKER_OPS_FILE="ops/docker-ops-${DOCKER_OPS_VERSION}"
+mkdir -p ops
+if [[ ! -f $DOCKER_OPS_FILE ]];then
+  wget --quiet -O $DOCKER_OPS_FILE https://github.com/kudulab/docker-ops/releases/download/${DOCKER_OPS_VERSION}/docker-ops
+fi
+source $DOCKER_OPS_FILE
+image_name="todolist-api/api"
+image_registry_prefix="todolist-api.apiregistry"
+image_registry="${image_registry_prefix}.azurecr.io"
```
Comment (mhurbuns) [inline tasks:13]: remove

**Review by mhurbuns — CHANGES_REQUESTED** [review]
Good start. Remove e2e references. 
We need build and test tasks

**Comment by pradhanangan** [comment]
> Good start. Remove e2e references. We need build and test tasks

@mhurbuns Updated code.

**Review by michaelm-clearpoint — DISMISSED** [review]

**Review by mniem — APPROVED** [review]
LGTM

**Pull request merged**